### PR TITLE
Feature/tao 10366 create event for changing resource privileges

### PR DESCRIPTION
**Pull request by andreluizmachado**
This PR is intended to implement the issue https://oat-sa.atlassian.net/browse/TAO-10366
Please also take a look at the PRs:
- https://github.com/oat-sa/tao-core/pull/2684
- https://github.com/oat-sa/lib-tao-elasticsearch/pull/34

**The issue:** 

When the user updates Data Access Control on the resource, these changes are not reflected in the advanced search.

**The solution:** 
A new event was created (DataAccessControlChangedEvent) to be possible to have Advanced search up to date

**Extra fixes**
- 500 error on search action (actions/class.Search.php) when trying to filter permissions on empty results (https://github.com/oat-sa/tao-core/pull/2684).
- Fail during search resources that have access control by the user instead of by role. We need to include the user identifier on the query to get a proper result. (https://github.com/oat-sa/lib-tao-elasticsearch/pull/34).
- fixed wrong interface implementation IndexSinceLastRunService:runIndexing method ((https://github.com/oat-sa/tao-core/pull/2684).

### CODE REVIEW

File: model/PermissionsService.php [hunk 1]
```diff
@@ -197,9 +198,10 @@ private function validateResources(array $resultPermissions): void
     /**
      * @param array $addRemove
      * @param string $resourceId
+     * @param bool $isRecursive
      */
 
-    private function triggerEvents(array $addRemove, string $resourceId): void
+    private function triggerEvents(array $addRemove, string $resourceId, bool $isRecursive): void
```
Comment (luisfn) [inline model/PermissionsService.php:204]: Is this the only call for triggerEvents? Not even from other extensions?

**Review by luisfn — COMMENTED** [review]
It looks good to me, my only question is about the call to triggerEvents method.

File: model/PermissionsService.php (hunk 1 above)
Comment (andreluizmachado) [reply model/PermissionsService.php:204]: Yes, I think that you did not see it, but the method is private.

**Review by luisfn — APPROVED** [review]
**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by Skyld — APPROVED** [review]
- [x] New code is covered by tests
- [ ] Tests are running successfully (old and new ones) on my local machine
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues
- [ ] Feature is working correctly on my local machine
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes
- [ ] Documentation is updated according to changes

**Pull request merged**